backends/backend_libkcapi.c

```diff
@@ -0,0 +1,2365 @@
+        kcapi_aead_setassoclen(handle, assoclen);
+
+        /* Convert CCM nonce into IV of size of 16 bytes */
+        ret = kcapi_aead_ccm_nonce_to_iv(data->iv.buf, data->iv.len, &newiv, &newivlen);
+        if (ret)
+        {
+                logger(LOGGER_ERR, "ccm: nonce conversion to IV failed\n");
+                goto out;
+        }
+
+        ret = -ENOMEM;
+        outbuflen = kcapi_aead_outbuflen_enc(handle, datalen, assoclen, taglen);
+        inbuflen = kcapi_aead_inbuflen_enc(handle, datalen, assoclen, taglen);
+        outbuf = calloc(1, outbuflen);
+        if (!outbuf)
+        {
+                logger(LOGGER_ERR, "ccm: allocation of buf failed\n");
+                goto out;
+        }
+
+        kcapi_aead_getdata_output(handle, outbuf, outbuflen, 1,
+                                        &assoc, &assoclen, &o_data, &datalen,
+                                        &tag, &taglen);
+
+        /* Seting key */
+        if (!data->key.len ||
+                !data->key.buf ||
+                kcapi_aead_setkey(handle, data->key.buf, data->key.len))
+        {
+                logger(LOGGER_ERR, "ccm: symmetric cipher setkey failed\n");
+                goto out;
+        }
+
+        if(data->assoc.buf)
+                memcpy(assoc, data->assoc.buf, assoclen);
+        if(data->data.buf)
+                memcpy(o_data, data->data.buf, datalen);
+
+        /* Performing encryption */
+        ret = kcapi_aead_encrypt(handle, outbuf, inbuflen,
+                                                newiv,
+                                                outbuf, outbuflen,
+                                                KCAPI_ACCESS_SENDMSG);
+        if (ret < 0)
+        {
+                logger(LOGGER_ERR, "ccm: cipher operation of buffer failed: %d %d\n", errno, ret);
+                goto out;
+        }
+        else if ((uint32_t)ret != outbuflen)
+        {
+                logger(LOGGER_ERR, "ccm: received data length %d does not match expected length %u\n", ret, outbuflen);
+                goto out;
+        }
+
+        if(data->iv.buf)
+                free_buf(&(data->iv));
+        if(data->data.buf)
+                free_buf(&(data->data));
+        if(data->tag.buf)
+                free_buf(&(data->tag));
+
+        CKINT_LOG(alloc_buf(datalen, &data->data), "ccm: ct buffer could not be allocated\n");
+
+        if(o_data)
+                memcpy(data->data.buf, o_data, datalen);
+
+        CKINT_LOG(alloc_buf(taglen, &data->tag), "ccm: tag buffer could not be allocated\n");
+
+        if(tag)
+                memcpy(data->tag.buf, tag, taglen);
+out:
+        kcapi_aead_destroy(handle);
+        if (newiv)
+                free(newiv);
+        if (outbuf)
+                free(outbuf);
+        return ret;
+}
+
+
+static int ccm_decrypt(struct aead_data *data, flags_t parsed_flags)
+{
+        struct kcapi_handle *handle = NULL;
+        char cipher[] = "ccm(aes)";
```

Comment:
As discussed, please use existing function to return the name, instead of hardcoding.